server/service/handler.go

```diff
@@ -874,6 +874,7 @@ func attachFleetAPIRoutes(r *mux.Router, svc fleet.Service, config config.FleetC
 	oeWindowsMDM.POST("/api/fleet/orbit/disk_encryption_key", postOrbitDiskEncryptionKeyEndpoint, orbitPostDiskEncryptionKeyRequest{})
 
 	oe.POST("/api/fleet/orbit/luks_data", postOrbitLUKSEndpoint, orbitPostLUKSRequest{})
+	oe.DELETE("/api/fleet/orbit/luks_data", deleteOrbitLUKSEndpoint, orbitDeleteLUKSRequest{})
```

Comment:
A simpler approach that wouldn't require a new API is:

- Define a new detail query (for linux devices only) that retrieves key slots and salts.
  - Would mean a new fleetd table for linux to return LUKS2 key slots + salts only (which may also be handy in the future for customers/users).
- Fleet can then process the returned rows and delete the escrowed keys if it's not present in the returned rows.

The detail query would run every 1 hour like the other detail queries, so we don't need to worry about this running too often. (Users can do a host refetch to trigger a check.)

Let me know your thoughts.

Reply:
Thanks. 

IMHO the more work we can offset to the host the better for performance reasons, but you make a good point about that having a new `luks` table will be useful in the long term, plus if we are doing other host checks using cron jobs I think we should keep things consistent,  so I'm happy to follow your suggested approach.

One thing that didn't cross my mind at the time I was implementing this is that adding a new route might require some configuration changes on client deployments (depending on their network setup), which I guess is one of the major cons of this approach :thinking: